flink-end-to-end-tests/test-scripts/elasticsearch-common.sh

```diff
@@ -37,20 +37,10 @@ function setup_elasticsearch {
     local elasticsearchDir=$TEST_DATA_DIR/elasticsearch
     mkdir -p $elasticsearchDir
     echo "Downloading Elasticsearch from $downloadUrl ..."
-    for i in {1..10};
-    do
-        wget "$downloadUrl" -O $TEST_DATA_DIR/elasticsearch.tar.gz
-        if [ $? -eq 0 ]; then
-            echo "Download successful."
-            echo "Extracting..."
-            tar xzf $TEST_DATA_DIR/elasticsearch.tar.gz -C $elasticsearchDir --strip-components=1
-            if [ $? -eq 0 ]; then
-                break
-            fi
-        fi
-        echo "Attempt $i failed."
-        sleep 5
-    done
+    retry_times_with_exponential_backoff 10 wget "$downloadUrl" -O $TEST_DATA_DIR/elasticsearch.tar.gz
+    echo "Download successful."
+    echo "Extracting..."
+    tar xzf $TEST_DATA_DIR/elasticsearch.tar.gz -C $elasticsearchDir --strip-components=1
```

Comment:
We should also retry the download in case the downloaded artifact is corrupt.

Reply:
When looking at the test failures, I've only seen instances where the connection to download/get data was interrupted. Both `curl` and `git` report those as errors which are picked up by the Bash script as in "something went wrong" (basically an exit code which is different then 0) and then it tries again. I haven't seen instances where the download has been reported as completed successfully and then extracting/using the artifacts fails. So should we do this hardening now?

Reply:
we could offer an optional hash argument, if the hash is not correct we do the the retry logic as well?

Reply:
Agreed, let's do that in a separate ticket and PR to update the `get_artifact` function to check if the download has been completed successfully.

Reply:
I've created https://issues.apache.org/jira/browse/FLINK-25125 for this